lib/msal-node/src/client/ManagedIdentitySources/AppService.ts

```diff
@@ -80,72 +85,55 @@ export class AppService extends BaseManagedIdentitySource {
         managedIdentityId: ManagedIdentityId
     ): ManagedIdentityRequestParameters {
         const request: ManagedIdentityRequestParameters =
-            new ManagedIdentityRequestParameters(HttpMethod.GET, this.endpoint);
+            new ManagedIdentityRequestParameters(
+                HttpMethod.GET,
+                this.identityEndpoint
+            );
+
+        request.headers[APP_SERVICE_SECRET_HEADER_NAME] = this.identityHeader;
+
+        request.queryParameters[API_VERSION_QUERY_PARAMETER_NAME] =
+            APP_SERVICE_MSI_API_VERSION;
+        request.queryParameters[RESOURCE_QUERY_PARAMETER_NAME] = resource;
+
+        if (
+            managedIdentityId.idType !== ManagedIdentityIdType.SYSTEM_ASSIGNED
```

Comment:
nit: `=== type.UserAssigned` seems to be more in line with your intention.

Reply:
that type doesn't exist. ManagedIdentityTypes looks like:
```typescript
const ManagedIdentityIdType = {
    SYSTEM_ASSIGNED: "system-assigned",
    USER_ASSIGNED_CLIENT_ID: "user-assigned-client-id",
    USER_ASSIGNED_RESOURCE_ID: "user-assigned-resource-id",
    USER_ASSIGNED_OBJECT_ID: "user-assigned-object-id",
}
```
Instead of using === for all three user assigned values, it's much less code to say ```!== ManagedIdentityIdType .SYSTEM_ASSIGNED```